### PR TITLE
Api removed from main

### PR DESCRIPTION
**Pull request by ryanahawthorne**
Fixed merge conflicts with typescript changes from main.
Moved api out from mail.js and now more dynamically grabs carousel data from the movie database

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -17,33 +15,31 @@ import {
...
+import { RowObjectType } from './types';
 
 // TODO move to helper functions module
 const easeOutQuint = (delta: number) => {
     return 1 - Math.pow(1 - delta, 5);
 }
 
 // BUG fast up and down causes glitch - prevDirection and a timer hack used until this is fixed properly
-// TODO move API stuff to seperate module
+// TODO add carousel row titles - show highlighted carousel row title to the right that fades out over time
```
Comment (iepsen) [inline src/main.ts:30]: It would be nice to have these things as issues here on the repo =)

File: src/api.ts [hunk 2]
```diff
@@ -0,0 +1,124 @@
+import { TARGET_POSITION_X, TARGET_POSITION_Y, IMAGE_WIDTH } from "./constants";
+import { PopularResponseType, RowImageType, RowObjectType } from './types';
+
+const authKey = 'c0140031611f36da79a4001affaad896';
+let genres: Array<RowObjectType> = [];
+// TODO these two funtions should be kept in a helpers class
+import { drawScaledCard, applyTranslate } from "./renderCards.js";
```
Comment (iepsen) [inline src/api.ts]:
```suggestion
import { drawScaledCard, applyTranslate } from "./renderCards";
```

File: src/api.ts [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+          }
+        })
+      })
+    });
+  })
+};
+
+
+// loads and re-renders images
+export const loadAndRenderImages = (rowsData: Array<RowObjectType>, ctx: CanvasRenderingContext2D, translateY: number ) => {
```
Comment (iepsen) [inline src/api.ts:105]: By the name of this function, I think we could file an issue with further refactoring here.

File: src/api.ts [hunk 4]
```diff
@@ -0,0 +1,124 @@
+import { TARGET_POSITION_X, TARGET_POSITION_Y, IMAGE_WIDTH } from "./constants";
+import { PopularResponseType, RowImageType, RowObjectType } from './types';
+
+const authKey = 'c0140031611f36da79a4001affaad896';
+let genres: Array<RowObjectType> = [];
```
Comment (iepsen) [inline src/api.ts:5]: It should be defined after all the imports

File: src/api.ts (hunk 3 above)
Comment (ryanahawthorne) [reply src/api.ts:105]: Yeah absolutely. I thought a fair bit about this and I don't have a simpler solution yet that doesn't add additional complexity.
As soon as the image is loaded - we need to update the image on the carousel to replace and blank spots.
Missing feature - the function should check if an animation is in progress, if so then rendering is not necessary as the new image will display on the next frame of the animation.

File: src/main.ts (hunk 1 above)
Comment (ryanahawthorne) [reply src/main.ts:30]: Great idea. Will do so.

**Pull request merged**